src/main/java/uk/ac/cam/cl/dtg/util/email/MailJetApiClientWrapper.java

```diff
@@ -0,0 +1,181 @@
+     * @throws MailjetException  - if underlying MailjetClient throws an exception
+     */
+    public void permanentlyDeleteAccountById(final String mailjetId) throws MailjetException {
+        Validate.notNull(mailjetId);
+        MailjetRequest request = new MailjetRequest(Contacts.resource, mailjetId);
+        MailjetResponse response = mailjetClient.delete(request);
+        int responseStatus = response.getStatus();
+        if (!(responseStatus == Response.Status.OK.getStatusCode())) {
+            // TODO: Do we want to get any of the data from this request?
+            throw new MailjetException("Failed to delete user!");
+        }
+    }
+
+    /**
+     *  Add a new user to MailJet or find by email if they already exist as a fallback
+     *
+     * @param email - email address
+     * @return the MailJet user ID
+     * @throws MailjetException  - if underlying MailjetClient throws an exception
+     */
+    public String addNewUser(final String email) throws MailjetException {
+        if (null == email) {
+            return null;
+        }
+        try {
+            MailjetRequest request = new MailjetRequest(Contact.resource).property(Contact.EMAIL, email);
+            MailjetResponse response = mailjetClient.post(request);
+            // Get MailJet ID out:
+            JSONObject responseData = response.getData().getJSONObject(0);
+            return Integer.toString(responseData.getInt("ID"));
+        } catch (MailjetClientRequestException e) {
+            if (e.getMessage().contains("already exists")) {
+                // FIXME - we need to test that this response always comes back with "already exists" in the message
+                log.warn(String.format("Attempted to create a user with email (%s) that already existed!", email));
+                JSONObject existingMailJetAccount = getAccountByIdOrEmail(email);
+                return Integer.toString(existingMailJetAccount.getInt("ID"));
+            } else {
+                log.error(String.format("Failed to create user in MailJet with email: %s", email), e);
+            }
+        } catch (JSONException e) {
+            log.error(String.format("Failed to create user in MailJet with email: %s", email), e);
+        }
+        return null;
+    }
+
+    /**
+     *  Update user details for an existing MailJet account
+     *
+     * @param mailjetId - MailJet user ID
+     * @throws MailjetException  - if underlying MailjetClient throws an exception
+     */
+    public void updateUserProperties(final String mailjetId, final String firstName, final String role,
+                                      final String email_verification_status) throws MailjetException {
+        Validate.notNull(mailjetId);
+        MailjetRequest request = new MailjetRequest(Contactdata.resource, mailjetId)
+                .property(Contactdata.DATA, new JSONArray()
+                        .put(new JSONObject().put("Name", "firstname").put("value", firstName))
```

Comment:
Again it would be nice in the future to have DTOs or something to help make this look a bit nicer? Although not urgent for now.

Reply:
Well their Java client requires `JSONObjects` in that format, so we'd have to make a DTO that could convert itself into a `JSONObject` of the right form. If we ever needed it in more than one place, it might be worth thinking about, but since each endpoint needs a different style of `JSONObject` so far it seemed unnecessary!